iDiff/utils/docker_utils.go

```diff
@@ -0,0 +1,172 @@
+package utils
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os/exec"
+	"regexp"
+	"strings"
+	"syscall"
+
+	img "github.com/docker/docker/api/types/image"
+	"github.com/docker/docker/client"
+	"github.com/golang/glog"
+)
+
+// ValidDockerVersion determines if there is a Docker client of the necessary version locally installed.
+func ValidDockerVersion() (bool, error) {
+	cli, err := client.NewEnvClient()
+	if err != nil {
+		return false, fmt.Errorf("Docker client error: %s", err)
+	}
+	version := cli.ClientVersion()
+	if version == "1.31" {
+		return true, nil
+	}
+	return false, nil
+}
+
+func GetImageHistory(image string) ([]img.HistoryResponseItem, error) {
+	imageID := image
+	var err error
+	var history []img.HistoryResponseItem
+	if !CheckImageID(image) {
+		imageID, _, err = pullImageCmd(image)
+		if err != nil {
+			return history, err
+		}
+	}
+	histArgs := []string{"history", "--no-trunc", imageID}
+	dockerHistCmd := exec.Command("docker", histArgs...)
+	var response bytes.Buffer
+	dockerHistCmd.Stdout = &response
+	if err := dockerHistCmd.Run(); err != nil {
```

Comment:
@r2d4 had mentioned that you might also be able to pull images using without using docker using something like https://github.com/google/containerregistry .  This could remove the docker dependency from the tool for CI.